Week3/server.js

```diff
@@ -0,0 +1,101 @@
+const bodyParser = require('body-parser');
+const express = require('express');
+const app = express();
+const db = require('./database.js');
+
+app.use(bodyParser.json());
+
+app.get('/user/:id', function (req, res) {
+  db.getAll(req.params.id).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error));
+});
+
+app.post('/create-user/:id', function (req, res) {
+  let id = req.params.id;
+  let firstName = req.body.firstName;
+  let lastName = req.body.lastName;
+  db.createUser(id, firstName, lastName).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.post('/create-list/:id', function (req, res) {
+  let UserId = req.params.id;
+  let category = req.body.category;
+  let prioritize = req.body.prioritize;
+  db.createList(UserId, category, prioritize).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.get('/get-lists/:id', function (req, res) {
+  db.getAll(req.params.id).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error));
+});
+
+app.post('/create-item/:id', function (req, res) {
+  let listId = req.params.id;
+  let todo = req.body.todo;
+  let description = req.body.description;
+  let prioritize = req.body.prioritize;
+  db.createItem(listId, todo, description, prioritize).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error.message));
+});
+
+app.get('/get-items/:id', function (req, res) {
+  db.getAll(req.params.id).then(result => {
+    res.send(result);
+  }).catch(error => console.log(error));
+});
+
+app.delete('/delete-user/:id', function (req, res) {
```

Comment:
When you make use of these methods ```delete()```, ```get()```, ```post()``` from express.js, you are binding the requests to HTTP METHODS. So for a cleaner API you could make the paths like this:

```javascript
app.get('/user/:id', function (req, res) {
  db.getAll(req.params.id).then(result => {
    res.send(result);
  }).catch(error => console.log(error));
});
app.delete('/user/:id', function (req,res) {let id = req.params.id;
  db.deleteUser(id).then(result => {
    res.send(result);
  }).catch(error => console.log(error.message));
});
```

And the client would request the same paths but with different HTTP METHODS. I added some extra links for more info too. =)

A link for HTTP METHODS:  https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods.
And here is a link for express.js methods doc: https://expressjs.com/en/api.html#app.delete.method